cpp/harvest/nacjd.cc

```diff
@@ -355,28 +385,41 @@ void ParseJSONAndWriteMARC(MARC::Writer * const title_writer) {
 
 
 int Main(int argc, char *argv[]) {
-    if (argc < 3 or argc > 4)
+    if (argc < 3)
         Usage();
 
     unsigned count_rows(9000);
-    if (argc == 4) {
-        count_rows = std::stoi(argv[3]);
+    unsigned mode_param(0);
+    std::string mode("");
+
+    if (argv[3]) {
+        mode = argv[3];
+    }
+
+    if (mode == "count_of_rows") {
+        count_rows = std::stoi(argv[4]);
     }
 
     auto marc_reader(MARC::Reader::Factory(argv[1]));
     auto marc_writer(MARC::Writer::Factory(argv[2]));
 
-    // parse marc_file (later phase_x) and store ids in set
-    std::set<std::string> parsed_marc_ids;
-    LOG_INFO("Extracting existing ICPSR ids from marc input...");
-    ExtractExistingIDsFromMarc(marc_reader.get(), &parsed_marc_ids);
-    LOG_INFO("Found: " + std::to_string(parsed_marc_ids.size()) + " records with ICPSR ids.");
-
-    // Download possible new publications and store in json file
-    LOG_INFO("Extracting ICPSR ids from website...");
-    unsigned number_of_new_ids;
-    ExtractIDsFromWebsite(parsed_marc_ids, &number_of_new_ids, count_rows);
-    LOG_INFO(std::to_string(number_of_new_ids) + " new ids collected from website.");
+    if (mode == "get_by_ID") {
+        mode_param = std::stoi(argv[4]);
```

Comment:
For each mode we must test first whether we have a suitable number of arguments (`argc`) and exit by calling `Usage()` if it does not exist, before we try to access `argv`.

Reply:
Also, for this case we want to call this variable `id` instead of `mode_param`.